tests/unit-tests/conversions.cpp

```diff
@@ -59,4 +51,107 @@ TEST_CASE("C++ Standard Conversions") {
       lexicon.double_type()
   );
   lexicon.make_if_then(condition, *lexicon.make_expr_stmt(then_expr));
+
+  // Bitcast                      (Pretend)
+  INFO("(bool*)ptr");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(ptr),
+    lexicon.get_pointer(lexicon.bool_type()),
+    lexicon.get_pointer(lexicon.bool_type())
+  );
+}
+
+TEST_CASE("Class Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  INFO("struct Base {};");
+  INFO("struct Derived : Base {};");
+  auto& base = *lexicon.make_class(*unit.global_region());
+  auto& derived = *lexicon.make_class(*unit.global_region());
+  derived.declare_base(base);
+
+  INFO("Base b;");
+  INFO("Derived d;");
+  auto& base_ptr = lexicon.get_pointer(base);
+  auto& derived_ptr = lexicon.get_pointer(derived);
+  auto& b = *unit.global_region()->declare_var(*lexicon.make_identifier("b"), base_ptr);
+  auto& d = *unit.global_region()->declare_var(*lexicon.make_identifier("d"), derived_ptr);
+
+  // Derived-to-base conversion
+  INFO("(Base*)b;");
+  lexicon.make_narrow(
+    *lexicon.make_id_expr(d),
+    base,
+    base_ptr
+  );
+
+  // Checked base-to-derived conversion
+  INFO("dynamic_cast<Derived*>(b);");
+  lexicon.make_widen(
+    *lexicon.make_id_expr(b),
+    derived,
+    derived_ptr
+  );
+
+  // Unchecked base-to-derived conversion
+  INFO("(Derived*)b;");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(b),
+    derived_ptr,
+    derived_ptr
+  );
+}
+
+TEST_CASE("CV Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  // Standard C++ CV qualification
+  INFO("(volatile int)7;");
+  lexicon.make_qualification(
+    *lexicon.make_literal(lexicon.double_type(), "2.2"),
+    Type_qualifier::Volatile,
+    lexicon.int_type() // prvalue will be adjusted to remove qualifiers (see 7.2.2/2)
+  );
+
+  INFO("const int* ptr;");
+  const auto& ptr_type = lexicon.get_qualified(
+    Type_qualifier::Const, lexicon.get_pointer(lexicon.int_type()));
+  auto& ptr = *unit.global_region()->declare_var(*lexicon.make_identifier("ptr"), ptr_type);
+
+  // Removal of const is non-standard C++ conversion
+  // Keep original explicit Const_cast
```

Comment:
What do you mean by "non-standard C++ conversion"?

Reply:
not a implicit conversion, will update comment.